protocols/collective_aggregation_protocol.go

```diff
@@ -195,11 +196,15 @@ func (p *CollectiveAggregationProtocol) Dispatch() error {
 
 // Announce forwarding down the tree.
 func (p *CollectiveAggregationProtocol) aggregationAnnouncementPhase() error {
-	dataReferenceMessage := <-p.DataReferenceChannel
-	if !p.IsLeaf() {
-		if err := p.SendToChildren(&dataReferenceMessage.DataReferenceMessage); err != nil {
-			return errors.New("Error sending <DataReferenceMessage>:" + err.Error())
+	select {
+	case dataReferenceMessage := <-p.DataReferenceChannel:
+		if !p.IsLeaf() {
+			if err := p.SendToChildren(&dataReferenceMessage.DataReferenceMessage); err != nil {
+				return errors.New("Error sending <DataReferenceMessage>:" + err.Error())
```

Comment:
You can use the new `fmt.Errorf` when returning error (it will also keep the stacktrace).

```suggestion
				return fmt.Errorf("Error sending <DataReferenceMessage>: %v", err)
```

Reply:
switched everywhere to fmt.Errorf: 8861342